### DISCUSSION TITLE
Exclude specific columns

### DISCUSSION THREAD

**Discussion by zsolt777** [dotnetcore/FreeSql#787] [Q&A]
Hi,

Could you tell me please if it is possible (or how) to include / exclude specific columns in select statement?

I have a lot of columns in a table and one of them contains large data that I do not want to retrieve all the times.

Thank you!
Zsolt

**Comment by 2881099** [comment]
```c#
class Song
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Descrption { get; set; }
}

class SongLite
{
    public Guid Id { get; set; }
    public string Name { get; set; }
}

var result = fsql.Select<Song>().ToList<SongLite>();
```

**Reply by zsolt777** [reply]
I'll keep in mind your solution, that is great, but in my case I have to use the "Song" class instead of "SongLite".

IgnoreColumns would be excellent like in updates:

`var result = fsql.Select<Song>().IgnoreColumns(a => new { a.Description }).ToList();
`

(btw: this is an awesome library)

Thank you!

**Reply by luoyunchong** [reply]
```
fsql.Select<Song>().ToList(a => new Song { Id = a.Id,Name=a.Name })
```
sql
```
select id,name from song
```

**Reply by zsolt777** [reply]
`fsql.Select<Song>().ToList(a => new Song { Id = a.Id,Name=a.Name })`

this is an excellent feature if you need just a few columns. But when you need 25-30 columns most of the times, then it is not good, moreover, if I add a new column to the table, then I should add the column to all of the selects as well.

I hope this will be implemented sometime:
`var result = fsql.Select<Song>().IgnoreColumns(a => new { a.Description }).ToList();`

Thanks anyway!
